workspace/pipeline/pln_horizon_to_odw.json

```diff
@@ -99,11 +99,11 @@
 							"type": "Expression"
 						},
 						"ErrorMessage": {
-							"value": "@replace(\n    concat(\n        activity('Record horizon to odw data loading')?.Error?.message, '|',\n        activity('Lookup_orchestration_file')?.Error?.message, '|',\n        activity('Filter_Horizon_Table')?.Error?.message, '|',\n        activity('Horizon Raw to Std')?.Error?.message, '|',\n        activity('Filter_Harmonised_NoteBook')?.Error?.message, '|',\n        variables('H_NoteBook_errorMessage')\n    ),\n    '|',\n    ''\n)\n",
+							"value": "@replace(\n    concat(\n        coalesce(activity('Record horizon to odw data loading')?.Error?.message, ''), '|',\n        coalesce(activity('Lookup_orchestration_file')?.Error?.message, ''), '|',\n        coalesce(activity('Filter_Horizon_Table')?.Error?.message, ''), '|',\n        coalesce(activity('Horizon Raw to Std')?.Error?.message, ''), '|',\n        coalesce(activity('Filter_Harmonised_NoteBook')?.Error?.message, ''), '|',\n        if(\n            not(equals(length(variables('HrmNotebookErrorArray')), 0))\n,\n            'horizon Harmonised NoteBook Failed check App Insights for details',\n            ''\n        )\n    ),\n    '|',\n    ''\n)\n",
```

Comment:
Also here the replace is a bit brokenish you're removing all char which means you lose all the seperatoirs and end up with one mashed string. Also it doesn't remove empty pieces cleanly it just deletes the delimiter
You should : 
- keep the pipes and just trim()
- or only concatenate non empty messages
- or use a delimiter that you don't later delete
But at least don't replace cause it destroys readability

Reply:
Same for this one - Make sure you change all expressions